### PR TITLE
 Implement RGBW correctly

### PR DESCRIPTION
**Pull request by Skaronator**

### CODE REVIEW

**Comment by xoseperez** [comment]
Current approach was that (when _light_use_whit) white channel was used when RGB channels had the same value. Your proposal is to use white always for the "common" (minimum) value of all three channels, and use the color LEDs for the rest. 

Does it make the difference? What is your testing hardware?

**Comment by Skaronator** [comment]
Yes it makes a difference. You can have the White LED turn on and add a little red for example. Also if you switch from white to a color it works much smoother since it doesn't instantly turn off the white strip.

My Hardware is a H801 with a RGB strip and a white strip. I'm ordering a CCT (Cold white + Warm white LED) Strip and a RGB strip once the color temperature is working properly.

I noticed a problem with the current approache: When you switch the color and go over the white spot it'll turn off the RGB strip and show white only. Technically this is correctly but doesn't look that good. Sadly, I don't have my Hue LED Strip anymore to compare it against Phillips implementation.

It might be better to calculate the RGBW values before the transition and not while the transition.

image

**Comment by Skaronator** [comment]
I've updated the RGB transition code. It looks much better now. I still need to implement this in the HSV functions.

And #695 should be pretty easy to implement now.

**Comment by xoseperez** [comment]
Fine

I'll try it before merging. I was concerned about breaking previous user experience with the light. Setting Use White to OFF will actually turn OFF all these changes, right?

**Comment by Skaronator** [comment]
If you turn off the White channel setting it doesn't change anything.

This is probably a better explanation for old code vs this PR:

image
_________
image

**Comment by Skaronator** [comment]
PR should be ready for testing. Works fine for me and I didn't spot any more bugs.

File: code/espurna/light.ino [hunk 1]
```diff
@@ -474,8 +474,16 @@ void _lightProviderUpdate() {
 // -----------------------------------------------------------------------------
 
 void _lightColorSave() {
+
+    //TODO: Remove this once this code was in a stable release
+    // This force set ch3 to 0 which could be not zero when you update from a old version
+    if (_light_use_white) {
+      setSetting("ch", 3, 0);
```
Comment (Skaronator) [inline code/espurna/light.ino]: @xoseperez I'm not sure about this. We need to reset this setting when a user upgrade from a old to a new version.

Should I leave it like this? Or do you have a script that run once a user flash/upgrade? Or does it wipe these settings anyway when you flash it?

Comment (Skaronator) [reply code/espurna/light.ino]: I've rewritten most of the code so this is no longer needed ;)

**Pull request merged**